docs/GettingStartedDocs/IntelSDKPortingGuideLinux.md

```diff
@@ -0,0 +1,311 @@
+`oe_sgx_enclave_properties_t::config.attributes` manually, even though they
+aren't supported explicitly by the `OE_SET_ENCLAVE_SGX` macro or *enclave*.conf
+file. Below is an example that grants the enclave access to `PROVISION_KEY`.
+
+```C
+OE_INFO_SECTION_BEGIN
+volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx = {
+    .header = {.size = sizeof(oe_sgx_enclave_properties_t),
+               .enclave_type = OE_ENCLAVE_TYPE_SGX,
+               .size_settings = {.num_heap_pages = 512,
+                                 .num_stack_pages = 512,
+                                 .num_tcs = 4}},
+    .config = {.product_id = 1234,
+               .security_version = 5678,
+               .attributes = OE_SGX_FLAGS_PROVISION_KEY |
+                             OE_MAKE_ATTRIBUTES(0)},
+    .end_marker = 0xecececececececec};
+OE_INFO_SECTION_END
+```
+
+If compatibilities with both SDKs are desired, avoid using features specific to
+either SDK.
+
+As a final note, neither Intel SGX SDK for Linux nor Open Enclave SDK provides
+configuration settings for enabling/disabling X features (e.g., AVX, AVX-512,
+etc.) explicitly. Open Enclave's SGX enclave loader uses the enabled X features
+on the local platform to initialize `SECS::ATTRIBUTES::XFRM`, and hard-codes
+`SIGSTRUCT::ATTRIBUTEMASK::XFRM` to `0`. That is, X features are *NOT* enforced
+and must *NOT* be relied upon for security.
+
+## Migrate ECall/OCall Definitions (EDL Files)
+
+Both Intel SGX SDK for Linux and Open Enclave SDK support the same grammer for
+defining trusted/untrusted functions (aka. ECalls/OCalls) in EDL files.
+However, built-in OCalls are defined in different headers. Intel's built-in
+OCalls are defined in `sgx_tstdc.edl` while Open Enclave SDK's are defined in
+`platform.edl`.
+
+Most EDL files include (by `include` statements) common C headers for both host
+and enclave sides. The most commonly included header is the one defining SGX
+architectural structures, which is `arch.h` in Intel SGX SDK for Linux or
+`openenclave/bits/sgx/sgxtypes.h` in Open Enclave SDK. Please also note that
+some structures may be named differently, e.g., the EINITTOKEN architectural
+structure is defined as `token_t` in Intel SGX SDK for Linux but `einittoken_t`
+in Open Enclave SDK.
+
+The code snippet below shows a way to include/import C headers and EDL
+definitions conditionally, in order to be compatible with both SDKs.
+
+```
+enclave {
+#ifdef OEEDGER8R
+    include "openenclave/bits/sgx/sgxtypes.h";
+    from "openenclave/edl/sgx/platform.edl" import *;
+#else
+    include "arch.h";
+    from "sgx_tstdc.edl" import *;
+#endif
+    /* ECall/OCall definitions go here */
+}
+```
+
+The macro `OEEDGER8R` used above could be defined by appending `-DOEEDGER8R` to
+`oeedger8r` command line. Please note that only `oeedger8r` supports defining
+macros at command line, even though both `sgx_edger8r` and `oeedger8r` support
+preprocessing.
```

Comment:
I would suggest being explicit and offering examples of both here:

````suggestion
For example, if using `oeedger8r`:

```bash
oeedger8r --trusted ${CMAKE_SOURCE_DIR}/MyEnclave.edl --search-path ${OE_INCLUDEDIR}) -DOEEDGER8R
```

`sgx_edger8r` also supports macro preprocessing in EDL files, but does not accept macro definitions as arguments. To use `sgx_edger8r`:
````

... you'll need to fill in an equivalent for sgx_edger8r as I'm unfamiliar with the syntax off the top of my head.